src/Integration/ProjectPropertyManager.cs

```diff
@@ -0,0 +1,123 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ProjectPropertyManager.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using EnvDTE;
+using System;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    [Export(typeof(IProjectPropertyManager))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class ProjectPropertyManager : IProjectPropertyManager
+    {
+        public const string TestProperty = Constants.SonarQubeTestProjectBuildPropertyKey;
+        public const string ExcludeProperty = Constants.SonarQubeExcludeBuildPropertyKey;
+
+        private readonly IProjectSystemHelper projectSystem;
+
+        [ImportingConstructor]
+        public ProjectPropertyManager(IHost host)
+        {
+            if (host == null)
+            {
+                throw new ArgumentNullException(nameof(host));
+            }
+
+            this.projectSystem = host.GetService<IProjectSystemHelper>();
+            this.projectSystem.AssertLocalServiceIsNotNull();
+        }
+
+        internal /*for testing purposes*/ ProjectPropertyManager(IProjectSystemHelper projectSystem)
+        {
+            this.projectSystem = projectSystem;
+        }
+
+        #region IProjectPropertyManager
+
+        public IEnumerable<Project> GetSelectedProjects()
+        {
+            return this.projectSystem?
```

Comment:
There's nothing stopping it from being null, only a debug assertion in the ctor. If we want to guarantee it's not null at construction then I should be throwing an exception.
In release this code will fail gracefully as is. As to whether this should fail with a big-bang or fail silently here is the question.

Just asserting an continuing on as if all is well is pretty poor. If it's not correct then it's a bug and wrong, and an Exception should be thrown at the point that it makes sense, in this case the ctor of the class, rather getting a NullRefException in an arbitrary method.